src/js/components/dashboard/graph/BarChartStacked.jsx

```diff
@@ -0,0 +1,352 @@
+/**
+ * BarChartStacked.jsx
+ * Created by Lizzie Salita 11/15/19
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { scaleLinear, scaleBand } from 'd3-scale';
+import { min, max } from 'lodash';
+
+import BarChartXAxis from './BarChartXAxis';
+
+/* eslint-disable react/no-unused-prop-types */
+// allow unused prop types. they are indirectly accessed as nextProps
+const propTypes = {
+    height: PropTypes.number,
+    width: PropTypes.number,
+    data: PropTypes.object,
+    padding: PropTypes.object,
+    legend: PropTypes.array,
+    showTooltip: PropTypes.func,
+    hideTooltip: PropTypes.func,
+    toggleTooltip: PropTypes.func
+};
+/* eslint-enable react/no-unused-prop-types */
+
+const defaultProps = {
+    padding: {
+        left: 50,
+        bottom: 50
+    }
+};
+
+export default class BarChartStacked extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            chartReady: false,
+            virtualChart: {}
+        };
+    }
+
+    componentDidMount() {
+        this.buildVirtualChart(this.props);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.data !== this.props.data) {
+            this.buildVirtualChart(this.props);
+        }
+        else if (prevProps.width !== this.props.width || prevProps.height !== this.props.height) {
+            this.buildVirtualChart(this.props);
```

Comment:
Why are these two different if statements? You call the same function with the same parameters